### PR TITLE
Patches of File API

### PR DESCRIPTION
**Pull request by izgzhen**
r? @Manishearth 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] There are tests for these changes OR

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12544)

### CODE REVIEW

**Comment by izgzhen** [comment]
Two simple fixes addressing:

1. The `/` in file name passed to `File` constructor.
2. Dummy FileID returned when the operation is invalid (e.g. https://w3c.github.io/FileAPI/#createRevokeMethodsParams)

The first fix is no problem; For the second one, I used an empty string before because first I am lazy and second it is easier to debug if something goes wrong. But that doesn't quite fit in with spec, though either an empty string or a random ID would have the same actual effect.

File: components/script/dom/file.rs [hunk 1]
```diff
@@ -73,7 +73,8 @@ impl File {
         let ref typeString = blobPropertyBag.type_;
 
         let modified = filePropertyBag.lastModified;
-        Ok(File::new(global, BlobImpl::new_from_bytes(bytes), filename, modified, typeString))
+        let replaced_filename = DOMString::from_string(filename.replace("/", ":"));
```
Comment (Manishearth) [inline components/script/dom/file.rs]: Link to the spec bug here

**Comment by Manishearth** [comment]
r=me with the link

File: components/script/dom/file.rs (hunk 1 above)
Comment (izgzhen) [inline components/script/dom/file.rs]: What do you mean by "spec bug"? Regarding link, this tiny replace operation is specified in the url above this function `pub fn Constructor` (which I unfortunately neglected at first)

Comment (Manishearth) [inline components/script/dom/file.rs]: Oh sorry, I was talking about https://github.com/w3c/FileAPI/issues/41 (the bug that wishes to remove this behavior)

File: components/script/dom/file.rs [hunk 2]
```diff
@@ -73,7 +73,10 @@ impl File {
         let ref typeString = blobPropertyBag.type_;
 
         let modified = filePropertyBag.lastModified;
-        Ok(File::new(global, BlobImpl::new_from_bytes(bytes), filename, modified, typeString))
+        // NOTE: Following behaviour might be removed in future,
+        // see https://github.com/w3c/FileAPI/issues/41
```
Comment (izgzhen) [inline components/script/dom/file.rs]: @Manishearth added

**Comment by Manishearth** [comment]
@bors-servo retry

looks intermittent

**Pull request merged**